### ISSUE TITLE
Need to create an index to use startAfterDocument and it does not work

### ISSUE THREAD

**Issue by creativecreatorormaybenot** [flutter/flutter#32031]
When trying to use [`startAfterDocument`](https://pub.dartlang.org/documentation/cloud_firestore/latest/cloud_firestore/Query/startAfterDocument.html) in **`cloud_firestore`**, I will be prompted to create an index. If I follow the link to create that index and wait for it to finish, i.e. that it is enabled and *should* work, it will still prompt me to create that index even though it is already created.

Through some digging I found out that this problem occurs because of [this line](https://github.com/flutter/plugins/blob/master/packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java#L218). Apparently, composite indexes with the documentID is not something that is supported.

To me it seems like it the issue is how the plugin stores the document snapshots. Maybe it should save the original snapshots to use the **actual** `startAfter(documentSnapshot`, `startAt(documentSnapshot` etc.

Basically, [this](https://github.com/flutter/flutter/issues/23618) does still not work. Maybe I am wrong, but have you confirmed that these methods actually work? They do not for me.

**Comment by Purus** [comment]
I can confirm on the same issue..
Using `startAfterDocument` throws exception asking to create index. But the index is already created.

Looking for your inputs for a fix.

**Comment by Navil** [comment]
I experience the same issue.

**Comment by creativecreatorormaybenot** [comment]
This is fixed on Android in https://github.com/flutter/plugins/pull/1591

**Comment by rockwotj** [comment]
This is not fixed in flutter/plugins#1591 I commented there on the issue with both that fix and the `{start,end}{After,At}Document` feature in general.

**Comment by creativecreatorormaybenot** [comment]
@rockwotj I understand what you are trying to say, however, there is no way to deny that this change works for me and it will work for anyone that experiences this issue. The way these queries are built in Flutter is fundamentally flawed.

**Comment by Mikkelet** [comment]
i have the same problem. It suggests I create a compound index (the error gives me a link). However the suggested index only uses one field and thus doesnt execute... what the hell?

**Comment by collinjackson** [comment]
Based on https://github.com/flutter/plugins/pull/1591#issuecomment-500408600 it sounds like @creativecreatorormaybenot may have a solution for this in the works.

**Comment by creativecreatorormaybenot** [comment]
https://github.com/flutter/plugins/pull/1720 works for me and all green lights 🎉
